_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+```sh
+# If you don't have homebrew
+# see https://brew.sh
+
+# Install the JavaScript tools you'll need
+brew install nodejs yarn
+
+# You'll want Visual Studio Code to work on this
+brew cask install visual-studio-code
+
+# Install the React Native CLI
+yarn global add react-native-cli
+
+# Create a new React Native
+# project called TrendingArtists
+react-native init TrendingArtists
+```
+
+</div>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+<p>You'll need <code>node</code> and <code>yarn</code> installed globally so you can run JavaScript and handle dependency management respectively.<p>
+
+<p>For working inside a JavaScript project, I'd strongly recommend using Microsoft's <a href='https://code.visualstudio.com'>Visual Studio Code</a>, it does a great job for React Native.</p>
+
+<p>Next up we're going to make the initial project and look around, so once all the installing has finished. You can follow along with the next section.</p>
+
+</div>
+</article>
+<article class='post'>
+
+Alright, so that should do a lot of downloading, and you'll have a new folder with a fully set up project for iOS and Android. We'll be focusing on the iOS side.
+
+So, open up `TrendingArtists` in your editor and inside your terminal with `cd TrendingArtists`. From the terminal you can get the Xcode project compiled, and opened inside the iOS simulator with `react-native start ios`. 
+
+This first uses `xcodebuild` to compile the native app found in `ios/TrendingArtists.xcodeproj` it will then load up the the React Native Packager. We'll cover that later, for now, think of it as a JavaScript file change watcher.
+
+Once a simulator has popped up, you'll see the "Welcome to React Native" screen. Now that we've got "an app" running. Let's take a moment to look through what we got.
+
+
+</article>
+<article class="desktop-only" style='display: flex; flex-flow: row;'>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+
+```sh
+$ tree .
+
+├── __tests__
+│   ├── index.android.js
+│   └── index.ios.js
+├── android - [snipped]
+├── app.json
+├── index.android.js
+├── index.ios.js
+├── ios
+│   ├── TrendingArtists
+│   │   ├── AppDelegate.h
+│   │   ├── AppDelegate.m
+│   │   ├── Base.lproj
+│   │   │   └── LaunchScreen.xib
+│   │   ├── Images.xcassets
+│   │   │   └── AppIcon.appiconset
+│   │   │       └── Contents.json
+│   │   ├── Info.plist
+│   │   └── main.m
+│   ├── TrendingArtists-tvOS
+│   │   └── Info.plist
+│   ├── TrendingArtists-tvOSTests
+│   │   └── Info.plist
+│   ├── TrendingArtists.xcodeproj
+│   │   ├── project.pbxproj
+│   │   ├── project.xcworkspace
+│   │   │   ├── contents.xcworkspacedata
+│   └── TrendingArtistsTests
+│       ├── Info.plist
+│       └── TrendingArtistsTests.m
+├── jsconfig.json
+├── package.json
+├── node_modules [x million files snipped]
+└── yarn.lock
+```
+
+</div>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+<p><strong>What are we looking at?</strong></p>
```

Comment:
After building+running the Xcode project, I had to wait a few minutes for the packager to do some work. Just a note.